applications/nrf_desktop/src/modules/fast_pair.c

```diff
@@ -119,6 +136,13 @@ static bool handle_module_state_event(const struct module_state_event *event)
 	return false;
 }
 
+static bool handle_ble_dongle_peer_event(const struct ble_dongle_peer_event *event)
+{
+	dongle_app_id = event->bt_app_id;
```

Comment:
I would disable provider if `dongle_app_id` is equal to selected application local identity